src/app/dashboard/create_budget_form.tsx

```diff
@@ -47,7 +52,51 @@ export function NewBudgetForm({user_id}: {user_id: string}) {
                 <TextInput
                   id="description"
                   onChange={e => setDescription(e.target.value)}
-                  value={description}
+                  value = {description}
+                  required
+                />
+              </div>
+              <div>
+               <div className="m-2 block">
+                  <Label htmlFor="event-location" value="Event Location" />
+                </div>
+                <TextInput
+                  id="event_location"
+                  onChange={e => setLocation(e.target.value)}
+                  value = {location}
+                  required
+                />
+              </div>
+              <div>
+               <div className="m-2 block">
+                  <Label htmlFor="event-date" value="Event Date" />
+                </div>
+                <TextInput
+                  id="event_date"
+                  onChange={e => setDate(e.target.value)}
+                  value = {date}
+                  required
+                />
+              </div>
+              <div>
+               <div className="m-2 block">
+                  <Label htmlFor="event-type" value="Event Type" />
+                </div>
+                <TextInput
+                  id="event_type"
+                  onChange={e => setType(e.target.value)}
+                  value = {type}
+                  required
+                />
+              </div>
+              <div>
+               <div className="m-2 block">
+                  <Label htmlFor="total-cost" value="Total Cost" />
+                </div>
+                <TextInput
+                  id="total_cost"
+                  onChange={e => setCost(e.target.value)}
+                  value = {cost}
                   required
                 />
               </div>
```

Comment:
`cost` should not be manually set by the user so we should remove this from the new budget form